wallet/commands.go

```diff
@@ -934,41 +940,85 @@ func (wal *Wallet) SetupAccountMixer(walletID int, walletPassphrase string, errC
 	}()
 }
 
-func (wal *Wallet) NewVSPD(walletID int, accountID int32) *dcrlibwallet.VSPD {
-	return wal.multi.NewVSPD("http://dev.planetdecred.org:23125", walletID, accountID)
-}
-
 // TicketPrice get ticket price
-func (wal *Wallet) TicketPrice(walletID int) string {
-	wall := wal.multi.WalletWithID(walletID)
-	pr, err := wall.TicketPrice()
+func (wal *Wallet) TicketPrice() (int64, string) {
+	pr, err := wal.multi.WalletsIterator().Next().TicketPrice()
 	if err != nil {
 		log.Error(err)
-		return ""
+		return 0, ""
 	}
-	return dcrutil.Amount(pr.TicketPrice).String()
+	return pr.TicketPrice, dcrutil.Amount(pr.TicketPrice).String()
 }
 
 // PurchaseTicket buy a ticket with given parameters
-func (wal *Wallet) PurchaseTicket(walletID int, accountID int32, tickets uint32, passphrase []byte, expiry uint32) ([]string, error) {
-	wall := wal.multi.WalletWithID(walletID)
-	request := &dcrlibwallet.PurchaseTicketsRequest{
-		Account:               uint32(accountID),
-		Passphrase:            passphrase,
-		NumTickets:            tickets,
-		Expiry:                uint32(wal.multi.GetBestBlock().Height) + expiry,
-		RequiredConfirmations: dcrlibwallet.DefaultRequiredConfirmations,
-	}
-	hashes, err := wall.PurchaseTickets(request, "")
-	if err != nil {
-		return []string{}, err
-	}
+func (wal *Wallet) PurchaseTicket(host string, walletID int, accountID int32, tickets uint32, passphrase []byte, errChan chan error) {
 	go func() {
 		var resp Response
+		wall := wal.multi.WalletWithID(walletID)
+		if wall == nil {
+			go func() {
+				errChan <- ErrIDNotExist
+			}()
+			return
+		}
+
+		vspd := wal.multi.NewVSPD(host, walletID, accountID)
+		_, err := vspd.GetInfo()
+		if err != nil {
+			go func() {
+				errChan <- err
+			}()
+			return
+		}
```

Comment:
This looks like a new vspd instance is created every time the PurchaseTicket method is called.